Composer/packages/client/src/pages/design/DebugPanel/TabExtensions/WatchTab/WatchTabContent.tsx

```diff
@@ -0,0 +1,336 @@
+import { CommandBar, ICommandBarStyles } from 'office-ui-fabric-react/lib/CommandBar';
+import {
+  DetailsList,
+  DetailsListLayoutMode,
+  IColumn,
+  SelectionMode,
+  Selection,
+  IObjectWithKey,
+  IDetailsRowProps,
+  DetailsRow,
+  IDetailsListStyles,
+  IDetailsRowStyles,
+  IDetailsHeaderStyles,
+} from 'office-ui-fabric-react/lib/DetailsList';
+import formatMessage from 'format-message';
+import get from 'lodash/get';
+import { ScrollablePane } from 'office-ui-fabric-react/lib/ScrollablePane';
+import { Sticky, StickyPositionType } from 'office-ui-fabric-react/lib/Sticky';
+import { TooltipHost } from 'office-ui-fabric-react/lib/Tooltip';
+import { CommunicationColors, FluentTheme } from '@uifabric/fluent-theme';
+
+import { DebugPanelTabHeaderProps } from '../types';
+import {
+  dispatcherState,
+  rootBotProjectIdSelector,
+  watchedVariablesState,
+  webChatTrafficState,
+} from '../../../../../recoilModel';
+import { WatchVariablePicker } from '../../WatchVariablePicker/WatchVariablePicker';
+import { getMemoryVariables } from '../../../../../recoilModel/dispatchers/utils/project';
+import { WatchDataPayload } from '../../WatchVariablePicker/utils/helpers';
+
+import { WatchTabObjectValue } from './WatchTabObjectValue';
+
+const toolbarHeight = 24;
+
+const unavailbleValue = css`
+  font-family: ${FluentTheme.fonts.small.fontFamily};
+  font-size: ${FluentTheme.fonts.small.fontSize};
+  font-style: italic;
+  height: 16px;
+  line-height: 16px;
+`;
+
+const primitiveValue = css`
+  font-family: ${FluentTheme.fonts.small.fontFamily};
+  font-size: ${FluentTheme.fonts.small.fontSize};
+  color: ${CommunicationColors.shade10};
+  height: 16px;
+  line-height: 16px;
+`;
+
+const watchTableStyles: Partial<IDetailsListStyles> = {
+  root: {
+    maxHeight: `calc(100% - ${toolbarHeight}px)`,
+  },
+  contentWrapper: {
+    overflowY: 'auto' as any,
+    // fill remaining space after table header row
+    height: 'calc(100% - 60px)',
+  },
+};
+
+const rowStyles: Partial<IDetailsRowStyles> = {
+  cell: { minHeight: 32, padding: '8px 6px' },
+  checkCell: {
+    height: 32,
+    minHeight: 32,
+    selectors: {
+      '& > div[role="checkbox"]': {
+        height: 32,
+      },
+    },
+  },
+  root: { minHeight: 32 },
+};
+
+const commandBarStyles: Partial<ICommandBarStyles> = { root: { height: toolbarHeight, padding: 0 } };
+const detailsHeaderStyles: Partial<IDetailsHeaderStyles> = {
+  root: {
+    paddingTop: 0,
+  },
+};
+
+const NameColumnKey = 'watchTabNameColumn';
+const ValueColumnKey = 'watchTabValueColumn';
+
+const watchTableLayout: DetailsListLayoutMode = DetailsListLayoutMode.justified;
+
+// Returns the specified property from the bot state trace if it exists.
+// Ex. getValueFromBotTraceMemory('user.address.city', trace)
+export const getValueFromBotTraceMemory = (
+  valuePath: string,
+  botTrace: Activity
+): { value: any; propertyIsAvailable: boolean } => {
+  const pathSegments = valuePath.split('.');
```

Comment:
When we are trying to access memory scope properties. For ex `dialog` or `conversation` as whole. The valuePath.split('.') just leads to one element and gets popped up. If the goal is to check if the path exists or not can we just not use 
```javascript
const valueOfProp = get(botTrace?.value, valuePath, undefined)
if(valueOfProp === undefined) then the propertyIsAvailable is false?
```

Reply:
I'm using `propertyIsAvailable` to distinguish between an `undefined` value and an `unavailable` value because the design has different treatments for each.

`undefined` meaning that the property could exist on the parent key, but it is not set
`unavailable` meaning that the property **could not and does not exist** because the property's parent path is not set

That being said, as far as I know, we do not offer a way for users to set a property to undefined in the form editor, because `"someProperty": undefined` is not actually valid JSON even though it is valid JavaScript.

If we treat `undefined` values as `unavailable` then we don't need both treatments, but that might be confusing to users because standard dev tools treat them differently:

![image](https://user-images.githubusercontent.com/3452012/123714134-9b42bb80-d82a-11eb-98f1-b90db51f6f25.png)

So I guess this all depends on what the design is: Do we want to call all undefined / unset values `unavailable`?

@mewa1024

Reply:
yes, the design was trying to match what users would expect from "Watch" in dev tools, namely, different treatments for "not available" and "undefined". (@sorgh, any opinions? I think you mentioned the not available/undefined difference earlier.)

Reply:
related but separate:

to Srinaath's point, there is actually a bug when trying to watch the root memory scopes themselves `user`, `conversation`, etc. and I am trying to fix that

Reply:
> yes, the design was trying to match what users would expect from "Watch" in dev tools, namely, different treatments for "not available" and "undefined". (@sorgh, any opinions? I think you mentioned the not available/undefined difference earlier.)

undefined is an accepted value, not available is for the gibberish user enters freely in the picker like:
user.name is defined by the user and until it's set, it's undefined
soroush.is.awesome is not available but it's true

Reply:
so would the distinction then be:

`undefined` is anything not set in a **defined memory scope** (user, conversation, customUserScope, etc.)
`not available` is anything outside one of these scopes